### ISSUE TITLE
`unsafe-undefined` ignores SAFETY comments on declarations without a type annotation

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#371]
# `unsafe-undefined` ignores SAFETY comments on declarations without a type annotation

- **Severity:** Medium (default-on rule; its own documented escape hatch doesn't work)
- **Status:** Verified by reproduction
- **Location:** `src/linter/rules/unsafe_undefined.zig:223-238`

## The bug

While walking up the parents of an `undefined` literal, the rule checks the declaration type and then the safety comment:

```zig
// unsafe_undefined.zig:223
if (do_type_check and i == 1) {
    const decl = ast.fullVarDecl(parent) orelse unreachable;
    const ty = decl.ast.type_node.unwrap() orelse break;   // <-- exits the loop entirely
    ...
}
if (has_safety_comment or (safety_comment_check and hasSafetyComment(ctx, ast.nodeMainToken(parent)))) return;
```

When the var decl has **no type annotation**, `orelse break` exits the parent loop, skipping the safety-comment check on the next line — and discarding a `has_safety_comment` that may already have been found. Execution falls through to the unconditional report at line 304.

## User impact

The rule's own help text says "Add a `SAFETY: <reason>` before this line explaining why this code is safe." Users who follow that advice on an untyped declaration still get the warning; the documented suppression mechanism silently doesn't work, forcing a `zlint-disable` directive or a redundant type annotation.

## Reproduction

```sh
cat > uu.zig <<'EOF'
pub fn main() void {
    // SAFETY: initialized before use
    const y = undefined;
    _ = y;
}
EOF
zlint uu.zig
# warning: unsafe-undefined: `undefined` is missing a safety comment
```

## Proposed fix

Only the *type check* is impossible without an annotation, not the comment check. Replace the loop-`break` with a fall-through so control reaches the safety-comment check:

```zig
if (do_type_check and i == 1) {
    const decl = ast.fullVarDecl(parent) orelse unreachable;
    if (decl.ast.type_node.unwrap()) |ty| {
        // existing whitelist checks on `ty`
    }
}
if (has_safety_comment or ...) return;
```

Add pass cases to the rule tests: SAFETY comment over `const y = undefined;` (no annotation) must not be reported.
